### ISSUE TITLE
"ps: Invalid process id: -opid=" output when daemon is started

### ISSUE THREAD

**Issue by akloeber** [syncany/syncany#489]
When the daemon is started there is a strange output during startup:
```
$ sy daemon start
Starting daemon: .ps: Invalid process id: -opid=
.. syncanyd (pid 12554).
```
The daemon itself is running fine. 

__Environment:__
Syncany 0.4.5-alpha
Mac OS X 10.10.3

**Comment by binwiederhier** [comment]
@akloeber Interesting. Seems that OSX doesn't like the `ps -p"$ID" -opid=` command. Can you try this instead in the `sy` file:

```
daemon_running() {
	ID=$(daemon_get_pid)
	
	if [ "$ID" == "-1" ]; then
		echo "0"
	elif kill -0 "$ID" 2> /dev/null; then         # <<< CHANGED LINE
		echo "1"
	else
		echo "0"
	fi	
}
```

If you don't know where that file is, run `which sy` and then edit it from there.

**Comment by akloeber** [comment]
@binwiederhier: Strangely enough I'm not able to reproduce this error anymore although yesterday it occurred several times.

I did some investigation and found out that `ps -p"$ID" -opid=` is supported on Mac OS X:
```
$ ps -p11435 -opid=
11435
```
The error is triggered if `$ID` is not set:
```
$ ps -p -opid=
ps: Invalid process id: -opid=
```

**Comment by akloeber** [comment]
If `~/.config/syncany/daemon.pid` exists but is __empty__ this error occurs:
```
$ sy daemon start
ps: Invalid process id: -opid=
Starting daemon: ps: Invalid process id: -opid=
.. syncanyd (pid 11971).
```
But then the error is logged __twice__ which was not the case when opened this issue, so in the moment when `sy daemon start` was executed it must have been missing. Looks like some kind of race condition during startup.

**Comment by akloeber** [comment]
Maybe the problem is in here:
```
		while [ "$DAEMON_RUNNING" == "0" -a $TRIES -gt 0 ]; do
			DAEMON_RUNNING=$(daemon_running)
			echo -n "."
		
			DAEMON_PID="$(daemon_get_pid)"
			TRIES=$(($TRIES-1))
			sleep 1
		done
```
If creation of PID file is not atomic it can be accessed before its content is populated.

**Comment by akloeber** [comment]
The following implementation should make `daemon_get_pid()`more robust:
```
#!/usr/bin/env bash

daemon_get_pid() {
  local PID

  if [ -e "$DAEMON_PIDFILE" ]; then
    PID=$(cat "$DAEMON_PIDFILE")
  fi

  if [ -n "$PID" ]; then
    echo $PID
  else
    echo -1
  fi
}

echo -n "5555" > pid.normal
echo -n " 6666 " > pid.blanks
echo -n "" > pid.empty

DAEMON_PIDFILE=pid.normal; echo \|$(daemon_get_pid)\|
DAEMON_PIDFILE=pid.blanks; echo \|$(daemon_get_pid)\|
DAEMON_PIDFILE=pid.empty; echo \|$(daemon_get_pid)\|
DAEMON_PIDFILE=pid.missing; echo \|$(daemon_get_pid)\|

rm pid.*
```
Output is:
```
$ ./test.sh 
|5555|
|6666|
|-1|
|-1|
```

**Comment by akloeber** [comment]
Maybe this has something to do with #439 which looks like a similar error.

**Comment by pimotte** [comment]
Yes, if we could find out why the pid file is empty in the first place, that would be better. Improving the parsing function would fix your issue already.

Would you like to create a pull request for this? (The relevant file is gradle/daemon/syncanyd.skel) If you created the commits that will give you the credit. If you don't feel like doing that for one reason or another, I'd be happy to commit the fix myself.

**Comment by pimotte** [comment]
Fixed by #492
